src/masternodes/mn_checks.cpp

```diff
@@ -2692,21 +2700,40 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
                         return Res::Err("Cannot payback this amount of loan for %s, either payback full amount or less than this amount!", loanToken->symbol);
             }
 
-            res = mnview.SubMintedTokens(loanToken->creationTx, subLoan);
-            if (!res)
-                return res;
+            if (height < Params().GetConsensus().FortCanningHillHeight || kv.first != DCT_ID{0})
+            {
+                res = mnview.SubMintedTokens(loanToken->creationTx, subLoan);
+                if (!res)
+                    return res;
+            }
 
             CalculateOwnerRewards(obj.from);
-            // subtract loan amount first, interest is burning below
-            res = mnview.SubBalance(obj.from, CTokenAmount{kv.first, subLoan});
-            if (!res)
-                return res;
 
-            // burn interest Token->USD->DFI->burnAddress
+            // sub amount from balance and burn interest Token->USD->DFI->burnAddress
             if (subInterest)
             {
-                LogPrint(BCLog::LOAN, "CLoanTakeLoanMessage(): Swapping %s interest to DFI - %lld, height - %d\n", loanToken->symbol, subInterest, height);
-                res = SwapToDFIOverUSD(mnview, kv.first, subInterest, obj.from, consensus.burnAddress, height);
+                if (height < Params().GetConsensus().FortCanningHillHeight || kv.first != DCT_ID{0})
+                {
+                    // subtract loan amount first, interest is burning below
+                    res = mnview.SubBalance(obj.from, CTokenAmount{tokenId, subLoan});
+                    if (!res)
+                        return res;
+
+                    LogPrint(BCLog::LOAN, "CLoanTakeLoanMessage(): Swapping %s interest to DFI - %lld, height - %d\n", loanToken->symbol, subInterest, height);
+                    res = SwapToDFIOverUSD(mnview, tokenId, subInterest, obj.from, consensus.burnAddress, height);
+                }
+                else
+                {
+                    CAmount subLoanInDFI = mnview.GetAmountInCurrency(subLoan, {"DFI","USD"}, false, true, true);
```

Comment:
Declare 3 variables
```
bool useNextPrice = false. requireLivePrice = true, reverseDirection = true;
````

Reply:
Here, reverse direction is not the way, so it is false. And all params are default, but maybe it is more readable to put them there.

Reply:
> Here, reverse direction is not the way, so it is false. And all params are default, but maybe it is more readable to put them there.

No, reverse direction is correct since we want to get back the amount in DFI from DUSD.

Reply:
Yes you are correct, I thought he was commenting on another line.